Tests/AWSSDKSwiftCoreTests/TracingTests.swift

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AWSSDKSwift open source project
+//
+// Copyright (c) 2017-2020 the AWSSDKSwift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AWSSDKSwift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@testable import AWSSDKSwiftCore
+import AWSTestUtils
+import Baggage
+import Instrumentation
+import NIO
+import NIOHTTP1
+import XCTest
+
+class TracingTests: XCTestCase {
+    private struct TestContext: AWSClient.Context {
+        public var baggage: BaggageContext
+
+        init(traceContext: TestTracer.Context) {
+            var baggage = BaggageContext()
+            baggage.test = traceContext
+            self.baggage = baggage
+        }
+    }
+
+    func testTracingDownstreamCall() {
+        // bootstrap tracer
+        let tracer = TestTracer()
+        InstrumentationSystem.bootstrap(tracer)
+
+        // create new trace
+        // TODO: not possible using TracingInstrument API, see https://github.com/slashmo/gsoc-swift-tracing/issues/137
+        let context = TestContext(traceContext: .init())
+
+        let awsServer = AWSTestServer(serviceProtocol: .json)
+        let elg = MultiThreadedEventLoopGroup(numberOfThreads: 1)
```

Comment:
Whats `elg` for?

Reply:
that was copied from another test :-) will rename